### PR TITLE
Add upstream_cx_overflow stat

### PR DESCRIPTION
**Pull request by junr03**

### CODE REVIEW

File: include/envoy/upstream/upstream.h [hunk 1]
```diff
@@ -140,6 +140,7 @@ class HostSet {
   COUNTER(upstream_cx_http2_total)                                                                 \
   COUNTER(upstream_cx_connect_fail)                                                                \
   COUNTER(upstream_cx_connect_timeout)                                                             \
+  COUNTER(upstream_cx_overflow)                                                        \
```
Comment (mattklein123) [inline include/envoy/upstream/upstream.h]: formatting

File: source/common/http/http1/conn_pool.cc [hunk 2]
```diff
@@ -72,9 +72,14 @@ ConnectionPool::Cancellable* ConnPoolImpl::newStream(StreamDecoder& response_dec
   }
 
   if (host_->cluster().resourceManager(priority_).pendingRequests().canCreate()) {
+    bool connections_canCreate =
```
Comment (mattklein123) [inline source/common/http/http1/conn_pool.cc]: variable naming incorrect

File: include/envoy/upstream/upstream.h (hunk 1 above)
Comment (junr03) [inline include/envoy/upstream/upstream.h]: fixed

File: source/common/http/http1/conn_pool.cc (hunk 2 above)
Comment (junr03) [inline source/common/http/http1/conn_pool.cc]: fixed

File: source/common/http/http1/conn_pool.cc [hunk 3]
```diff
@@ -72,9 +72,14 @@ ConnectionPool::Cancellable* ConnPoolImpl::newStream(StreamDecoder& response_dec
   }
 
   if (host_->cluster().resourceManager(priority_).pendingRequests().canCreate()) {
+    bool connections_can_create =
```
Comment (mattklein123) [inline source/common/http/http1/conn_pool.cc]: nit: can_create_connection

**Pull request merged**